### DISCUSSION TITLE
Add Unsupervised Continual Anomaly Detection with Contrastively-learned Prompt

### DISCUSSION THREAD

**Discussion by samet-akcay** [open-edge-platform/anomalib#2362] [Feature Requests]
### Discussed in https://github.com/openvinotoolkit/anomalib/discussions/1940

<div type='discussions-op-text'>

# Description
This is to implement the following paper:

🧾 [Paper](https://ojs.aaai.org/index.php/AAAI/article/view/28153)
👩‍💻 [Code](https://github.com/shirowalker/UCAD/tree/main)

**Comment by samet-akcay** [comment]
@inbasperu can you comment here as well so I could assign this to you. Otherwise, Github does not let me do that

**Comment by inbasperu** [comment]
.take

**Comment by ashwinvaidya17** [comment]
@inbasperu are you still working on this?

**Comment by inbasperu** [comment]
Hey @ashwinvaidya17,

I do not have a CUDA-enabled device, so I wasn't able to test the author's code. I tried porting it to an MPS version, but due to unsupported types, I wasn't able to run it. Would it be possible to give me a week of time? I have made some arrangements to get access to a CUDA-enabled machine.

**Comment by ashwinvaidya17** [comment]
@inbasperu Yeah that's alright. I was just ensuring all issue are up-to-date
